plugin/metrics/prometheus/metricsstore/reader_test.go

```diff
@@ -85,14 +102,16 @@ func TestNewMetricsReaderInvalidAddress(t *testing.T) {
 func TestGetMinStepDuration(t *testing.T) {
 	params := metricsstore.MinStepDurationQueryParameters{}
 	logger := zap.NewNop()
+	tracer, exp := tracerProvider(t)
 	listener, err := net.Listen("tcp", "localhost:")
 	require.NoError(t, err)
 	assert.NotNil(t, listener)
 
-	reader, err := NewMetricsReader(logger, config.Configuration{
+	reader, err := NewMetricsReader(config.Configuration{
 		ServerURL:      "http://" + listener.Addr().String(),
 		ConnectTimeout: defaultTimeout,
-	})
+	}, logger, tracer)
+	assert.NotEmpty(t, exp.GetSpans(), "No spans recorded during the test.")
```

Comment:
why do you expect spans created at this point? There is no instrumentation in `NewMetricsReader` function. You know which functions do have it, it's in this PR, so only assert in tests for those functions.

Reply:
Now I got that. I found three but they don't invoke `NewMetricsReader` function. e.g. `TestGetLatencies` only tests against query svc.

Reply:
`TestGetLatencies` calls `prepareMetricsReaderAndServer` which calls `NewMetricsReader`

Reply:
On it.